code/modules/lighting/light_source.dm

```diff
@@ -126,76 +126,89 @@
 		lum_g = 1
 		lum_b = 1
 
-/*
-/datum/light_source/proc/falloff(atom/movable/lighting_overlay/O)
-  writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/datum/light_source/proc/falloff() called tick#: [world.time]")
-  #if LIGHTING_FALLOFF == 1 // circular
-	. = (O.x - source_turf.x)**2 + (O.y - source_turf.y)**2 + LIGHTING_HEIGHT
-   #if LIGHTING_LAMBERTIAN == 1
-	. = CLAMP01((1 - CLAMP01(sqrt(.) / light_range)) * (1 / (sqrt(. + 1))))
-   #else
-	. = 1 - CLAMP01(sqrt(.) / light_range)
-   #endif
-
-  #elif LIGHTING_FALLOFF == 2 // square
-	. = abs(O.x - source_turf.x) + abs(O.y - source_turf.y) + LIGHTING_HEIGHT
-   #if LIGHTING_LAMBERTIAN == 1
-	. = CLAMP01((1 - CLAMP01(. / light_range)) * (1 / (sqrt(.)**2 + )))
-   #else
-	. = 1 - CLAMP01(. / light_range)
-   #endif
+#if LIGHTING_FALLOFF == 1 //circular
+  #define LUM_DISTANCE(swapvar, O, T) swapvar = (O.x - T.x)**2 + (O.y - T.y)**2 + LIGHTING_HEIGHT
+  #if LIGHTING_LAMBERTIAN == 1
+    #define LUM_ATTENUATION(swapvar) swapvar = CLAMP01((1 - CLAMP01(sqrt(swapvar) / max(1,light_range))) * (1 / sqrt(swapvar + 1)))
+  #else
+    #define LUM_ATTENUATION(swapvar) swapvar = 1 - CLAMP01(sqrt(swapvar) / max(1,light_range))
   #endif
-*/
-
+#elif LIGHTING_FALLOFF == 2 //square
+  #define LUM_DISTANCE(swapvar, O, T) swapvar = abs(O.x - T.x) + abs(O.y - T.y) + LIGHTING_HEIGHT
+  #if LIGHTING_LAMBERTIAN == 1
+    #define LUM_ATTENUATION(swapvar) swapvar = CLAMP01((1 - CLAMP01(swapvar / max(1,light_range))) * (1 / sqrt(swapvar**2 + 1)))
+  #else
+    #define LUM_ATTENUATION(swapvar) swapvar = CLAMP01(swapvar / max(1,light_range))
+  #endif
+#endif
+
+#define LUM_FALLOFF(swapvar, O, T) \
+  LUM_DISTANCE(swapvar, O, T); \
+  LUM_ATTENUATION(swapvar);
+
+#define APPLY_TURF(T)						\
+	if(T.lighting_overlay)					\
```

Comment:
jesus H christ

you do realize that this ends up just copypasting the code wherever you use APPLY_TURF(T) right